src/lucky_flow/find_element.cr

```diff
@@ -13,27 +13,21 @@ class LuckyFlow::FindElement
   end
 
   def run
-    matching_elements.first? || retry
-  end
+    loop do
+      matching_elements = find_matching_elements
+      return matching_elements.first if matching_elements.first?
 
-  private def retry
-    self.tries += 1
-    if has_retries_left?
-      retry_after_delay
-    else
-      raise_element_not_found_error
+      break unless has_retries_left?
+      sleep retry_delay_in_ms
     end
+
+    raise_element_not_found_error
   end
 
   private def has_retries_left?
     tries < max_tries
   end
 
-  private def retry_after_delay
-    sleep retry_delay_in_ms
-    run
```

Comment:
Is the moral of this story that recursion is bad?  Hmmm... 🙈

Reply:
Haha yeah I guess so!